lib/cocoapods/generator/xcconfig/pod_xcconfig.rb

```diff
@@ -71,6 +71,7 @@ def generate
           XCConfigHelper.add_target_specific_settings(target, @xcconfig)
           recursive_dependent_targets = target.recursive_dependent_targets
           @xcconfig.merge! XCConfigHelper.search_paths_for_dependent_targets(target, recursive_dependent_targets, @test_xcconfig)
+          XCConfigHelper.generate_vendored_build_settings(target, recursive_dependent_targets, @xcconfig, false)
```

Comment:
should this always happen for all pod targets? For example what if someone does not use `use_frameworks!` and instead use static libs? Does the same apply?

Reply:
Good catch! I added the check. 

It also changed the back the different order on the `adds settings for test dependent targets excluding the parents targets` unit test to being unchanged from master